pkg/bindings/artifacts/extract.go

```diff
@@ -0,0 +1,78 @@
+package artifacts
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/podman/v5/pkg/bindings"
+)
+
+func Extract(ctx context.Context, artifactName string, targetDir string, options *ExtractOptions) error {
+	conn, err := bindings.GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if options == nil {
+		options = new(ExtractOptions)
+	}
+
+	params, err := options.ToParams()
+	if err != nil {
+		return err
+	}
+
+	response, err := conn.DoRequest(ctx, nil, http.MethodGet, "/artifacts/%s/extract", params, nil, artifactName)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if !response.IsSuccess() {
+		return response.Process(nil)
+	}
+
+	tr := tar.NewReader(response.Body)
```

Comment:
I have no big experience with external go modules, their maintainability and so on, but this tar handling code looks somehow low-level to me. Couldn't we use archiver/v3 instead?

My gut feeling tells me someone already did this code safely with path traversals and support for different tar features. What do the others think?

Update: while writing this comment, I checked what the other projects use and it seems tar.NewReader is used quite a lot, so maybe it is really the way to go. I will still keep my comment here for others to think, but the current way is probably fine...